core/src/main/scala/kanaloa/reactive/dispatcher/queue/Queue.scala

```diff
@@ -58,9 +59,7 @@ trait Queue extends Actor with ActorLogging with MessageScheduler {
       finish(status, s"Queue successfully retired")
       PartialFunction.empty //doesn't matter after finish, but is required by the api.
     } else handleWork(status, retiring) orElse {
-      case Enqueue(_, replyTo, _) ⇒
-        replyTo.getOrElse(sender) ! Retiring
-
+      case Enqueue         ⇒ sender() ! Retiring
```

Comment:
Something else just popped into my head while I was fixing this.  Should the `Retiring` object just be an `EnqueueRejected.reason` instance?  If I send a message to a Queue whilst it retires, I probably want a handle on the original message, identical to as if an EnqueueFailed.